### PR TITLE
8261072: AArch64: Fix MacroAssembler::get_thread convention

### PR DESCRIPTION
**Pull request by AntonKozlov**
Please review a fix in a special calling convention for aarch64_get_thread_helper for non-Linux platforms (windows/aarch64 for now). 

Preliminary review: https://mail.openjdk.java.net/pipermail/aarch64-port-dev/2021-January/011239.html

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261072](https://bugs.openjdk.java.net/browse/JDK-8261072): AArch64: Fix MacroAssembler::get_thread convention

### Reviewers
 * [Bernhard Urban-Forster](https://openjdk.java.net/census#burban) (@lewurm - Author) ⚠️ Review applies to 217d3c23e1bf81647fcdd18ceebe3213e2476866
 * [Andrew Haley](https://openjdk.java.net/census#aph) (@theRealAph - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2451/head:pull/2451`
`$ git checkout pull/2451`

### CODE REVIEW

**Review by lewurm — APPROVED** [review]

**Comment by VladimirKempik** [comment]
Looks good, will make other non-linux platform work without the need to change this ever again

File: src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp [hunk 1]
```diff
@@ -5266,10 +5266,14 @@ void MacroAssembler::char_array_compress(Register src, Register dst, Register le
 // by the call to JavaThread::aarch64_get_thread_helper() or, indeed,
 // the call setup code.
 //
-// aarch64_get_thread_helper() clobbers only r0, r1, and flags.
+// On Linux, aarch64_get_thread_helper() clobbers only r0, r1, and flags.
+// On Windows, the helper is a usual C function.
```
Comment (theRealAph) [inline src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp:5270]: This should say "other systems", not "Windows". Otherwise OK.

Comment (AntonKozlov) [reply src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp]: Thanks, fixed.

**Review by theRealAph — APPROVED** [review]

**Pull request closed without merging**